lib/couchbase/collection_options.rb

```diff
@@ -191,7 +190,7 @@ def exists?(path_or_index)
           end
         return false unless field
 
-        raise field.error unless field.error.nil?
+        raise field.error unless field.error.nil? || field.error.is_a?(Error::PathNotFound)
```

Comment:
The core can return `path_not_found` status codes if those are coming from `get` or `count` operations. Those should not be raised if `#exists` is called - they should only be raised for `#content`